sharding/mainchain/smc_client_test.go

```diff
@@ -1,6 +1,192 @@
 package mainchain
 
-import "github.com/ethereum/go-ethereum/sharding"
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/sharding"
+	"github.com/ethereum/go-ethereum/sharding/contracts"
+)
+
+var (
+	key, _                   = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	addr                     = crypto.PubkeyToAddress(key.PublicKey)
+	accountBalance1001Eth, _ = new(big.Int).SetString("1001000000000000000000", 10)
+)
 
 // Verifies that SMCCLient implements the sharding Service inteface.
 var _ = sharding.Service(&SMCClient{})
+
+// mockClient is struct to implement the smcClient methods for testing.
+type mockClient struct {
+	smc         *contracts.SMC
+	depositFlag bool
+	t           *testing.T
+	backend     *backends.SimulatedBackend
+	blockNumber *big.Int
+}
+
+// Mirrors the function in the main file, but instead of having a client to perform rpc calls
+// it is replaced by the simulated backend.
+func (m *mockClient) WaitForTransaction(ctx context.Context, hash common.Hash, durationInSeconds int64) error {
```

Comment:
How about just passing in a time.Duration?
https://golang.org/pkg/time/#Duration